### PR TITLE
rough version of fixing issue with  descriptions

### PR DESCRIPTION
**Pull request by dzhaniivanov**
https://github.com/podkrepi-bg/frontend/issues/1420

## Motivation and context

first version if u think its okey i will separate the logic  in component

i test it with big dummy text and i think its looks good on desktop and on mobile.
i am open to any suggestions :)

### CODE REVIEW

**Review by slavcho — CHANGES_REQUESTED** [review]
Great!
Test it on my machine - it works well:

image


Just fix the minor comments and push.

File: src/components/client/campaigns/CampaignPublicExpensesGrid.tsx [hunk 1]
```diff
@@ -96,6 +99,50 @@ export default observer(function CampaignPublicExpensesGrid({ slug }: Props) {
...
+
+        const open = Boolean(anchorEl)
+        return (
+          <div>
+            <Typography
+              aria-owns={open ? 'mouse-over-popover' : undefined}
+              aria-haspopup="true"
+              onMouseEnter={handlePopoverOpen}
+              onMouseLeave={handlePopoverClose}>
+              {params?.value.slice(0, 10)}...
```
[the remark this answers is not in the document]
Comment (slavcho) [reply src/components/client/campaigns/CampaignPublicExpensesGrid.tsx:122]: And remove the ... postfix

**Comment by sashko9807** [comment]
MUI Datagrid also supports variable row height based on the content [link to docs](https://mui.com/x/react-data-grid/row-height/#dynamic-row-height). This way the longer text descriptions should be visible, without any trim or ellipsis applied.

Note: I am just mentioning an alternative way to solve this issue. This is not a review

**Comment by slavcho** [comment]
> MUI Datagrid also supports variable row height based on the content [link to docs](https://mui.com/x/react-data-grid/row-height/#dynamic-row-height). This way the longer expense description should be visible, without any trim or ellipsis applied.
> 
> Note: I am just mentioning an alternative way to solve this issue. This is not a review

This is actually very good!
Lets do this - just tested it - works:

image

Dzhani - just add getRowHeight={() => 'auto'} - and remove the rest :)

**Comment by sashko9807** [comment]
> > MUI Datagrid also supports variable row height based on the content [link to docs](https://mui.com/x/react-data-grid/row-height/#dynamic-row-height). This way the longer expense description should be visible, without any trim or ellipsis applied.
> > Note: I am just mentioning an alternative way to solve this issue. This is not a review
> 
> This is actually very good! Lets do this - just tested it - it works:
> 
> 
> Dzhani - simply add getRowHeight={() => 'auto'} - and remove the rest :)

Consider adding min width to either the whole table or the description column, otherwise the text could still be clipped on smaller viewports(<600px), due to the smaller column width.

Another thing looking at the screenshot above,  maybe we should align the attachment files on a single row, rather than  column(if possible).

**Comment by dzhaniivanov** [comment]
for some reason getRowHeight doesn't work as we expected on my machine

**Comment by dzhaniivanov** [comment]
i changed the popover with auto getRowHeight prop but on my machine (test with chrome and firefox) it doesn't work as expected in the MUI docs. I (and Sashko) researched the possible problem and cleared the cache files, node_modules, re-cloned the repo but nothing worked. Slavcho, if you have any suggestion, please ping  me in discord maybe :)

**Review by igoychev — APPROVED** [review]

**Pull request merged**